src/wp-includes/option.php

```diff
@@ -900,7 +902,7 @@ function update_option( $option, $value, $autoload = null ) {
 	 *
 	 * See https://core.trac.wordpress.org/ticket/38903
 	 */
-	if ( $value === $old_value || maybe_serialize( $value ) === maybe_serialize( $old_value ) ) {
+	if ( sprintf( '%s', $serialized_value ) === $old_value || maybe_serialize( $old_value ) === $serialized_value ) {
```

Comment:
I am concerned about this, for the same reason that https://core.trac.wordpress.org/ticket/22192 didn't work out the way we were hoping. There is definitely room for BC breakage here.

`$old_value` is not necessarily a string or non-scalar value. The return value of `get_option()` can be filtered, and it's not mandated anywhere to return a string. And even without custom filters, this can commonly be another type, particularly when the default is returned that is specified via `register_setting()` and injected via core's own filter usage.

Reply:
Also, why are we using `$serialized_value` in the first clause? `$old_value` may be non-scalar in which case this is never `true`.

Could this maybe work instead?
```suggestion
	if (
		$value === $old_value ||
		( is_scalar( $value ) && is_scalar( $old_value ) && sprintf( '%s', $value ) === sprintf( '%s', $old_value ) ) ||
		maybe_serialize( $old_value ) === $serialized_value ) {
```

This seems a bit safer to me as it maintains both of the original clauses and only considers the problem case as a new clause for scalar values.